website/moneybirdsynchronization/models.py

```diff
@@ -389,3 +397,80 @@ def to_moneybird(self):
     class Meta:
         verbose_name = _("moneybird payment")
         verbose_name_plural = _("moneybird payments")
+
+
+class MoneybirdGeneralJournalDocument(models.Model):
+    order = models.OneToOneField(
+        Order,
+        on_delete=models.CASCADE,
+        verbose_name=_("order"),
+    )
+
+    moneybird_general_journal_document_id = models.CharField(
+        verbose_name=_("moneybird general journal document id"),
+        max_length=255,
+        blank=True,
+        null=True,
+    )
+
+    external_invoice = models.OneToOneField(
+        MoneybirdExternalInvoice,
+        on_delete=models.DO_NOTHING,
+        verbose_name=_("external invoice"),
+        related_name="moneybird_journal_external_invoice",
+        blank=True,
+        null=True,
+    )
+
+    needs_synchronization = models.BooleanField(
+        default=True,  # The field is set False only when it has been successfully synchronized.
+        help_text="Indicates that the journal has to be synchronized (again).",
+    )
+
+    needs_deletion = models.BooleanField(
+        default=False,
+        help_text="Indicates that the journal has to be deleted from moneybird.",
+    )
+
+    def __str__(self):
+        return f"Moneybird journal for {self.order}"
+
+    def to_moneybird(self):
```

Comment:
This is very specific to merchandise item bookings, while the model itself is a "general" model for all sorts of General Journal Documents. Right now we only use them for merchandise, but it might be nice to either rename it to explicitly be called MerchandiseJournalDocuments, or subclass it or give it some field that specifies it is about merchandise. So that later, in case we might want to sync different kinds of general journal documents as well, we don't have problems

Reply:
Im in favor of subclassing it btw